src/formatDate.js

```diff
@@ -8,7 +8,51 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(/[-/.]/);
+  const valueMap = fromFormat.reduce((accumulator, format, index) => {
+    accumulator[format] = dateParts[index];
+
+    return accumulator;
+  }, {});
+
+  const yy = Number(valueMap['YY']);
+  const year =
+    valueMap['YYYY'] ||
+    (yy >= 0 && yy <= 99
+      ? yy < 30
+        ? `20${String(yy).padStart(2, '0')}`
+        : `19${String(yy).padStart(2, '0')}`
+      : valueMap['YY']);
+
+  const month = valueMap['MM'] ? Number(valueMap['MM'], 10) - 1 : 0;
```

Comment:
It seems you are attempting to adjust the month value by subtracting 1, which is unnecessary for the task at hand. The task requires reformatting the date string, not creating a `Date` object. This line should be removed to avoid confusion.